### ISSUE TITLE
Support for WiFi add on devices

### ISSUE THREAD

**Issue by AkhilRam7** [mozilla-iot/http-on-off-mkr1000#2]
mkr1000 being a inbuilt Wi-Fi development board its quite easy to run this code. we should look at the thing where every board can be able to connect with WOT. We need to use software serial or other serial ports if available in order to work with the boards where wifi module is an add on

**Comment by dhylands** [comment]
Yeah - ideally, what should happen is that something like an MKR1000 can be a web thing directly and actually implement the REST API required. This code (and the corresponding adapter: https://github.com/mozilla-iot/http-on-off-adapter was just a quick hack I put together in a couple hours for a demo one of my team members was going to be doing.

**Comment by kgiori** [comment]
The quick hack request is my fault. I agree with Dave that this example should be changed to support the native web thing API for an onOffLight. The ability to have even little 8-bit MCUs connect as WoT devices is being tackled as a binary serial protocol. See first thoughts on that in the [serial-adapter](https://github.com/mozilla-iot/serial-adapter) repo.
